Bug: 并发测试报错

### 描述问题

https://github.com/he0119/nonebot-plugin-user/blob/85390ca895eb6d0f3aaba380a2896acab07bd7a2/tests/test_concurrency.py#L45

如果删除这一行，则会在测试中随机报错。

### 复现步骤

运行测试。

### 期望的结果

没有随机报错。

### 截图或日志

https://github.com/he0119/nonebot-plugin-user/actions/runs/15066934640/job/42353781473?pr=167

[12_Test (3.12, ubuntu-latest).txt](https://github.com/user-attachments/files/20364100/12_Test.3.12.ubuntu-latest.txt)

```log
05-16 19:01:02 [WARNING] nonebot_plugin_orm | 跳过启动检查, 正在同步数据库模式...
05-16 19:01:02 [DEBUG] alembic | Context impl SQLiteImpl.
05-16 19:01:02 [DEBUG] alembic | Will assume non-transactional DDL.
05-16 19:01:02 [DEBUG] sqlalchemy | BEGIN (implicit)
05-16 19:01:02 [DEBUG] sqlalchemy | PRAGMA main.table_info("alembic_version")
05-16 19:01:02 [DEBUG] sqlalchemy | [raw sql] ()
05-16 19:01:02 [DEBUG] sqlalchemy | PRAGMA temp.table_info("alembic_version")
05-16 19:01:02 [DEBUG] sqlalchemy | [raw sql] ()
05-16 19:01:02 [DEBUG] sqlalchemy | PRAGMA main.table_info("alembic_version")
05-16 19:01:02 [DEBUG] sqlalchemy | [raw sql] ()
05-16 19:01:02 [DEBUG] sqlalchemy | PRAGMA temp.table_info("alembic_version")
05-16 19:01:02 [DEBUG] sqlalchemy | [raw sql] ()
05-16 19:01:02 [DEBUG] sqlalchemy | 
CREATE TABLE alembic_version (
	version_num VARCHAR(32) NOT NULL, 
	CONSTRAINT alembic_version_pkc PRIMARY KEY (version_num)
)


05-16 19:01:02 [DEBUG] sqlalchemy | [no key 0.00022s] ()
05-16 19:01:02 [DEBUG] sqlalchemy | SELECT name FROM sqlite_master WHERE type='table' AND name NOT LIKE 'sqlite~_%' ESCAPE '~' ORDER BY name
05-16 19:01:02 [DEBUG] sqlalchemy | [raw sql] ()
05-16 19:01:02 [DEBUG] alembic | Detected added table 'nonebot_plugin_user_bind'
05-16 19:01:02 [DEBUG] alembic | Detected added table 'nonebot_plugin_user_user'
05-16 19:01:02 [DEBUG] alembic | Detected added table 'orm_test'
05-16 19:01:02 [DEBUG] alembic | Context impl DefaultImpl.
05-16 19:01:02 [DEBUG] alembic | Will assume non-transactional DDL.
05-16 19:01:02 [DEBUG] sqlalchemy | 
CREATE TABLE nonebot_plugin_user_bind (
	platform VARCHAR(32) NOT NULL, 
	platform_id VARCHAR(64) NOT NULL, 
	bind_id INTEGER NOT NULL, 
	original_id INTEGER NOT NULL, 
	CONSTRAINT pk_nonebot_plugin_user_bind PRIMARY KEY (platform, platform_id)
)


05-16 19:01:02 [DEBUG] sqlalchemy | [no key 0.00025s] ()
05-16 19:01:02 [DEBUG] sqlalchemy | 
CREATE TABLE nonebot_plugin_user_user (
	id INTEGER NOT NULL, 
	name VARCHAR(255) NOT NULL, 
	created_at DATETIME NOT NULL, 
	CONSTRAINT pk_nonebot_plugin_user_user PRIMARY KEY (id), 
	UNIQUE (name)
)


05-16 19:01:02 [DEBUG] sqlalchemy | [no key 0.00026s] ()
05-16 19:01:02 [DEBUG] sqlalchemy | 
CREATE TABLE orm_test (
	id INTEGER NOT NULL, 
	bot_id VARCHAR NOT NULL, 
	CONSTRAINT pk_orm_test PRIMARY KEY (id)
)


05-16 19:01:02 [DEBUG] sqlalchemy | [no key 0.00024s] ()
05-16 19:01:02 [DEBUG] sqlalchemy | COMMIT
05-16 19:01:02 [DEBUG] tests | test_permission_concurrency-查询 Test 表 0
05-16 19:01:02 [DEBUG] sqlalchemy | BEGIN (implicit)
05-16 19:01:02 [DEBUG] sqlalchemy | SELECT orm_test.id, orm_test.bot_id 
FROM orm_test
05-16 19:01:02 [DEBUG] sqlalchemy | [generated in 0.00029s] ()
05-16 19:01:02 [DEBUG] sqlalchemy | ROLLBACK
05-16 19:01:02 [DEBUG] tests | test_permission_concurrency-提交 Test 表
05-16 19:01:02 [SUCCESS] nonebot | OneBot V11 test | [message.group.normal]: Message 32 from 10@[群:10000] '/orm'
05-16 19:01:02 [DEBUG] nonebot | Checking for matchers in priority 1...
05-16 19:01:02 [DEBUG] nonebot | Plugin-Alconna | Alconna::orm 对 "/orm" 的解析结果是 (matched=True, header_match=(origin='/orm' result='/orm' matched=True groups={}))
05-16 19:01:02 [INFO] nonebot | Event will be handled by AlconnaMatcher(type='', module=tests.plugins.orm, lineno=18)
05-16 19:01:02 [DEBUG] nonebot | Running AlconnaMatcher(type='', module=tests.plugins.orm, lineno=18)
05-16 19:01:02 [DEBUG] nonebot | Running handler Dependent(call=handle_orm)
05-16 19:01:02 [DEBUG] sqlalchemy | BEGIN (implicit)
05-16 19:01:02 [DEBUG] sqlalchemy | SELECT nonebot_plugin_user_user.id, nonebot_plugin_user_user.name, nonebot_plugin_user_user.created_at 
FROM nonebot_plugin_user_user JOIN nonebot_plugin_user_bind ON nonebot_plugin_user_user.id = nonebot_plugin_user_bind.bind_id 
WHERE nonebot_plugin_user_bind.platform_id = ? AND nonebot_plugin_user_bind.platform = ?
05-16 19:01:02 [DEBUG] sqlalchemy | [generated in 0.00031s] ('10', 'QQClient')
05-16 19:01:02 [DEBUG] sqlalchemy | BEGIN (implicit)
05-16 19:01:02 [DEBUG] sqlalchemy | SELECT orm_test.id, orm_test.bot_id 
FROM orm_test
05-16 19:01:02 [DEBUG] sqlalchemy | [cached since 0.02184s ago] ()
05-16 19:01:02 [DEBUG] sqlalchemy | BEGIN (implicit)
05-16 19:01:02 [DEBUG] sqlalchemy | INSERT INTO nonebot_plugin_user_user (name, created_at) VALUES (?, ?)
05-16 19:01:02 [DEBUG] sqlalchemy | [generated in 0.00034s] ('QQClient-10', '2025-05-16 11:01:02.095410')
05-16 19:01:02 [DEBUG] sqlalchemy | INSERT INTO orm_test (bot_id) VALUES (?)
05-16 19:01:02 [DEBUG] sqlalchemy | [generated in 0.00033s] ('test',)
05-16 19:01:02 [DEBUG] sqlalchemy | COMMIT
05-16 19:01:02 [DEBUG] sqlalchemy | BEGIN (implicit)
05-16 19:01:02 [DEBUG] sqlalchemy | INSERT INTO nonebot_plugin_user_bind (platform, platform_id, bind_id, original_id) VALUES (?, ?, ?, ?)
05-16 19:01:02 [DEBUG] sqlalchemy | [generated in 0.00029s] ('QQClient', '10', 1, 1)
05-16 19:01:02 [DEBUG] sqlalchemy | COMMIT
05-16 19:01:02 [INFO] nonebot | AlconnaMatcher(type='', module=tests.plugins.orm, lineno=18) running complete
05-16 19:01:02 [DEBUG] sqlalchemy | COMMIT
05-16 19:01:02 [DEBUG] sqlalchemy | SELECT nonebot_plugin_user_user.id AS nonebot_plugin_user_user_id, nonebot_plugin_user_user.name AS nonebot_plugin_user_user_name, nonebot_plugin_user_user.created_at AS nonebot_plugin_user_user_created_at 
FROM nonebot_plugin_user_user 
WHERE nonebot_plugin_user_user.id = ?
05-16 19:01:02 [DEBUG] sqlalchemy | [generated in 0.00028s] (1,)
05-16 19:01:02 [DEBUG] nonebot | Checking for matchers completed
05-16 19:01:02 [DEBUG] tests | test_permission_concurrency-查询 Test 表 1
05-16 19:01:02 [DEBUG] sqlalchemy | BEGIN (implicit)
05-16 19:01:02 [DEBUG] sqlalchemy | SELECT orm_test.id, orm_test.bot_id 
FROM orm_test
05-16 19:01:02 [DEBUG] sqlalchemy | [cached since 0.03934s ago] ()
05-16 19:01:02 [DEBUG] sqlalchemy | ROLLBACK
F05-16 19:01:02 [DEBUG] sqlalchemy | BEGIN (implicit)
05-16 19:01:02 [DEBUG] sqlalchemy | DELETE FROM nonebot_plugin_user_bind
05-16 19:01:02 [DEBUG] sqlalchemy | [generated in 0.00028s] ()
05-16 19:01:02 [DEBUG] sqlalchemy | DELETE FROM nonebot_plugin_user_user
05-16 19:01:02 [DEBUG] sqlalchemy | [generated in 0.00030s] ()
05-16 19:01:02 [DEBUG] sqlalchemy | COMMIT
```

```log
=================================== FAILURES ===================================
_________________________ test_permission_concurrency __________________________

app = <nonebug.app.App object at 0x7fd2fc36d760>

    async def test_permission_concurrency(app: App):
        """测试权限和其他响应器同时访问数据库"""
        from nonebot_plugin_orm import _engines, get_session
    
        from tests.plugins.orm import orm_cmd
    
        engine = _engines[""].sync_engine
    
        engine.logger.info("test_permission_concurrency-查询 Test 表 0")
    
        async with get_session() as session:
            from tests.plugins.orm import Test
    
            test = await session.scalars(select(Test))
    
            assert len(test.all()) == 0
    
        engine.logger.info("test_permission_concurrency-提交 Test 表")
    
        async with app.test_matcher() as ctx:
            adapter = get_adapter(Adapter)
            bot = ctx.create_bot(base=Bot, adapter=adapter)
    
            event = fake_group_message_event_v11(message=Message("/orm"))
            ctx.receive_event(bot, event)
            ctx.should_call_api("get_group_info", {"group_id": 10000}, {})
            ctx.should_call_api("get_group_member_info", {"group_id": 10000, "user_id": 10, "no_cache": True}, {})
            ctx.should_call_send(event, "已提交！", None)
            ctx.should_finished(orm_cmd)
    
        engine.logger.info("test_permission_concurrency-查询 Test 表 1")
    
        async with get_session() as session:
            from tests.plugins.orm import Test
    
            test = await session.scalars(select(Test))
    
>           assert len(test.all()) == 1
E           assert 0 == 1
E            +  where 0 = len([])
E            +    where [] = all()
E            +      where all = <sqlalchemy.engine.result.ScalarResult object at 0x7fd2fc0c41e0>.all

tests/test_concurrency.py:75: AssertionError
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -23,7 +23,6 @@ def pytest_collection_modifyitems(items: list[pytest.Item]):
 
 def pytest_configure(config: pytest.Config) -> None:
     config.stash[NONEBOT_INIT_KWARGS] = {
-        "sqlalchemy_database_url": "sqlite+aiosqlite:///:memory:",
         "alembic_startup_check": False,
     }
 
@@ -43,7 +42,7 @@ async def app(app: App, mocker: MockerFixture, tmp_path: Path):
     nonebot.require("tests.plugins.orm")
     from nonebot_plugin_orm import get_session, init_orm
 
-    mocker.patch("nonebot_plugin_orm._data_dir", tmp_path / "orm")
+    mocker.patch("nonebot_plugin_orm._data_dir", tmp_path)
     # 确保 _insert_mutex 是在当前事件循环中创建的
     mocker.patch("nonebot_plugin_user.utils._insert_mutex", asyncio.Lock())
 

--- a/tests/test_concurrency.py
+++ b/tests/test_concurrency.py
@@ -38,7 +38,6 @@ async def test_permission_concurrency(app: App):
     """测试权限和其他响应器同时访问数据库"""
     from nonebot_plugin_orm import get_session
 
-    from nonebot_plugin_user import get_user
     from tests.plugins.orm import orm_cmd
 
     async with get_session() as session:

--- a/tests/test_concurrency.py
+++ b/tests/test_concurrency.py
@@ -41,9 +41,6 @@ async def test_permission_concurrency(app: App):
     from nonebot_plugin_user import get_user
     from tests.plugins.orm import orm_cmd
 
-    # FIXME: 不这样做的话，现在这个测试会随机报错
-    await get_user("QQClient", "10")
-
     async with get_session() as session:
         from tests.plugins.orm import Test
 

